ql/src/test/queries/clientpositive/stats_part_multi_insert.q

```diff
@@ -0,0 +1,18 @@
+-- Test multi inserting to the same partition
+
+create table source(p int, key int,value string);
+insert into source(p, key, value) values (101,42,'string42');
+
+create table stats_part(key int,value string) partitioned by (p int);
+
+from source
+insert into stats_part select key, value, p
+insert into stats_part select key, value, p;
+
+select p, key, value from stats_part;
+desc formatted stats_part;
+select count(*) from stats_part;
```

Comment:
To keep the test functional even if the default value of `hive.compute.query.using.stats` changes in the future I would add an explicit set to true just before the select query.